src/zenml/model/model_config.py

```diff
@@ -0,0 +1,214 @@
+    save_models_to_registry: bool = True
+    recovery: bool = False
+
+    @validator("create_new_model_version")
+    def _validate_create_new_model_version(
+        cls, create_new_model_version: bool, values: Dict[str, Any]
+    ) -> bool:
+        if create_new_model_version:
+            if values.get("version", None) is not None:
+                raise ValueError(
+                    "`version` cannot be used with `create_new_model_version`."
+                )
+        return create_new_model_version
+
+    @validator("recovery")
+    def _validate_recovery(
+        cls, recovery: bool, values: Dict[str, Any]
+    ) -> bool:
+        if recovery:
+            if not values.get("create_new_model_version", False):
+                logger.warning(
+                    "Using `recovery` flag without `create_new_model_version=True` makes no effect"
+                )
+        return recovery
+
+    @validator("version")
+    def _validate_version(
+        cls, version: Union[str, ModelStages]
+    ) -> Union[str, ModelStages]:
+        if isinstance(version, str) and version in ModelStages._members():
+            logger.warning(
+                f"Version `{version}` matches one of the possible `ModelStages`, if you want to fetch "
+                "model version by its' stage make sure to pass in instance of `ModelStages`."
+            )
+        return version
+
+    @property
+    def _stage(self) -> Optional[str]:
+        if isinstance(self.version, ModelStages):
+            return self.version.value
+        return None
+
+    def _get_request_params(
+        self,
+        request_model: Union[
+            Type[ModelRequestModel], Type[ModelVersionRequestModel]
+        ],
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        zenml_client = Client()
+        request_params = {
+            k: v
+            for k, v in self.dict().items()
+            if k in request_model.schema()["properties"]
+        }
+        request_params.update(kwargs)
+        request_params.update(
+            {
+                "user": zenml_client.active_user.id,
+                "workspace": zenml_client.active_workspace.id,
+            }
+        )
+        return request_params
+
+    def get_or_create_model(self) -> ModelResponseModel:
+        """This method should get or create a model from Model WatchTower.
+
+        New model is created implicitly, if missing, otherwise fetched.
+
+        Returns:
+            The model based on configuration.
+        """
+        zenml_client = Client()
+        try:
+            model = zenml_client.zen_store.get_model(
+                model_name_or_id=self.name
+            )
+        except KeyError:
+            model_request = ModelRequestModel.parse_obj(
+                self._get_request_params(ModelRequestModel)
+            )
+            model = zenml_client.zen_store.create_model(model=model_request)
+            logger.warning(f"New model `{self.name}` was created implicitly.")
```

Comment:
I added some safety-belt logic into this, not sure how it can fit SQLStore, especially since it can be running towards REST. I can introduce a get_or_create interface, with transactional logic, but it sounds like an overkill.

Reply:
Yeah exactly, we would need a `get_or_create` interface, but as you said it's overkill

Reply:
I'll just resolve this, IMO not worth the required effort